src/ShopBundle/objects/db/TShopVoucher.class.php

```diff
@@ -272,6 +272,14 @@ public function AllowUseOfVoucher()
                     $bAllowUse = TdbShopVoucher::USE_ERROR_CUSTOMER_USED_VOUCHER_SERIES_BEFORE;
                 }
             }
+
+            $matchingVoucherFromSameSeries = $oExistingVouchers->FindItemWithProperty(
```

Comment:
The actual field `fieldRestrictToOnePerUser` must be checked?

I would place this more to the front (after the "series inactive" check). It is a sort of early exit.

Reply:
yes fieldRestrictToOnePerUser must be checked. So I put it in the  same block as the one used as the user check. Maybe i can move the whole block to top with early return.